python/paddle/fluid/backward.py

```diff
@@ -888,6 +888,18 @@ def _append_backward_ops_with_checkpoints_(
                     continue
                 if name not in var_name_dict:
                     var_name_dict[name] = name + var_suffix
+
+                    # we should create the rename var in subprog, otherwise its VarType will be BOOL
+                    block.create_var(
+                        name=var_name_dict[name],
+                        shape=block.program.global_block().var(name).shape,
```

Comment:
`block.program.global_block().var(name)` is called four times.

Reply:
revised ~